src/store/river.js

```diff
@@ -26,15 +26,18 @@ export default function reducer(state=initialState, action) {
   switch (action.type) {
     case a.ADD_POST_TO_RIVER: {
       state = state.unshift(action.post.id);
+
       break;
     }
 
     case a.SET_POSTS_TO_RIVER: {
-      let posts = i.List(action.posts.map(post => post.id));
+      let posts = action.posts.map(post => post.id);
 
-      if (!i.is(state, posts)) {
-        state = posts;
-      }
+      posts.forEach(postID => {
+        if (!state.includes(postID)) {
+          state = state.push(postID);
+        }
+      });
```

Comment:
this wouldn't work in some cases. logic should be more complex. For example: you have 3 pages of contents loaded. then you navigate to other pages and return back to river. `fetchData` triggers loading of the first page of river (as it should). It has new posts (because your friends posted something new). 

This code means that New posts would ALWAYS go to the end of the list. that's wrong